pkg/apis/hive/v1alpha1/clusterstate_types.go

```diff
@@ -0,0 +1,51 @@
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	configv1 "github.com/openshift/api/config/v1"
+)
+
+// ClusterStateSpec defines the desired state of ClusterState
+type ClusterStateSpec struct {
+	ClusterDeployment corev1.LocalObjectReference `json:"clusterDeployment"`
+}
+
+// ClusterStateStatus defines the observed state of ClusterState
+type ClusterStateStatus struct {
+	ClusterOperators []ClusterOperatorState `json:"clusterOperators,omitempty"`
+}
+
+// ClusterOperatorState summarizes the status of a single cluster operator
+type ClusterOperatorState struct {
+	Name       string                                    `json:"name"`
+	Conditions []configv1.ClusterOperatorStatusCondition `json:"conditions,omitempty"`
+}
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ClusterState is the Schema for the clusterstates API
+// +k8s:openapi-gen=true
+// +kubebuilder:subresource:status
+type ClusterState struct {
```

Comment:
With this name it feels like it might be a nice home for ClusterVersionStatus that we currently store in ClusterDeployment.Status. In theory that may not be used and we could also move it here.

I like that these are done by two controllers, but then that's two controllers writing to the same type fairly often, I'm not sure if that's a good idea or not.

If we think it's ok to store cluster version status here someday (assuming it's a safe breaking API change), and possibly other state we might fetch from cluster in future, then I really like this naming.

If we don't like the sound of the above then this should probably be named something more specific like ClusterOperatorState.

Reply:
lol, the only reason I didn't name it ClusterOperatorState is that I didn't know what to name the thing that represents the status of a single ClusterOperator :)

Reply:
I think it makes sense that cluster version status be stored in this CRD and not on the clusterdeployment